src/module/actor/character/sheet.ts

```diff
@@ -754,6 +754,35 @@ class CharacterSheetPF2e<TActor extends CharacterPF2e> extends CreatureSheetPF2e
             }
         }
 
+        // Update all "normal" spellcasting entries
+        for (const select of htmlQueryAll<HTMLSelectElement>(html, "select[data-action=update-spellcasting-rank]")) {
+            select.addEventListener("click", () => {
+                const newRank = Number(select.value);
+                if ([1, 2, 3, 4].includes(newRank)) {
+                    const entries = this.actor.itemTypes.spellcastingEntry.filter((e) => !e.system.proficiency.slug);
+                    this.actor.updateEmbeddedDocuments(
+                        "Item",
+                        entries.map((e) => ({ _id: e.id, "system.proficiency.value": newRank })),
+                    );
+                }
+            });
+        }
+
+        // Update a single spellcasting entry that uses an alternative statistic
+        for (const select of htmlQueryAll<HTMLSelectElement>(
+            castingPanel,
```

Comment:
why castingPanel instead of html like the other?

Reply:
These are only in the spellcasting panel

Reply:
then should the other one also refer to castingPanel?